Server/Application/WalletMethods/Command/DeletWalletCommand.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using MediatR;
+using Endava_Project.Server.Data;
+using Microsoft.EntityFrameworkCore;
+using Endava_Project.Server.Helpers;
+using System.Threading;
+
+namespace Endava_Project.Server.Application.WalletMethods.Command
+{
+    public class DeletWalletCommand : IRequest<CommandResult>
+    {
+        public string UserId { get; set; }
+        public Guid WalletId { get; set; }
+    }
+
+    public class DeletWalletCommandHandler : IRequestHandler<DeletWalletCommand, CommandResult>
+    {
+        private readonly ApplicationDbContext context;
+
+        public DeletWalletCommandHandler(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<CommandResult> Handle(DeletWalletCommand command, CancellationToken cancellationToken)
+        {
+            
+            var user = context.Users.Include(x => x.Wallets).FirstOrDefault(x => x.Id == command.UserId);
```

Comment:
what if user is null?

Reply:
At next if (!user.Wallets.Any(x => x.Id == command.WalletId)) the comand will return CommandResult.ReturnFailure , then the controller will return BadRequest.

Reply:
`if (!user.Wallets.Any(x => x.Id == command.WalletId))` will throw NullReferenceException, because user is null. Which will map to 500 in your controller

Reply:
`user == null || !user.Wallets.Any(x => x.Id == command.WalletId)` is this a good idea?

Reply:
How about this:
We get the wallet only:
```csharp
var wallet = await context.Users
    .Where(u => u.Id == command.UserId)
    .SelectMany(u => u.Wallets)
    .FirstOrDefaultAsync(w => w.Id == command.WalletId);
```
If wallet is null => failure
Otherwise: `context.Wallets.Remove(wallet);`

This should work I think. Check what sql this query generates though.
Anyway, this is a bit more advanced, it's good as it is.